File: Libraries/Megamark Unity/Assets/Megamark Controller/Demo Scripts/Lasers.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

/*
 * This example checks the range of two attached laser rangefinders.
 * The collected data is then printed via the Processing Console in real-time.
 */

/*** NOTE: THIS EXAMPLE DOES NOT WORK, AS UNITY'S SERIAL.READLINE COMMAND FREEZES UNITY! ***/

public class Lasers : MonoBehaviour {

	public Megamark megamark;
	private IEnumerator coroutine;

	// Use this for initialization
	void Start () {
		coroutine = GetLasers();
		StartCoroutine(coroutine);
	}

	// Get both laser rangefinder data and print them to the console window every second.  
	IEnumerator GetLasers() {
		while (true) {
			// Get the left range...
			int left = megamark.getLeftLaser();
			// Get the right range...
			int right = megamark.getRightLaser();
			// Convert both to strings and print to console.
			Debug.Log("Left Laser: " + left + " | Right Laser: " + right);
			yield return new WaitForSeconds (2);
		}
	}
		
}

File: Libraries/Megamark RobotC/Lasers.c
#pragma config(CircuitBoardType, typeCktBoardMega)
#pragma config(Sensor, anlg3,  leftLaser,      sensorAnalog)
#pragma config(Sensor, anlg4,  rightLaser,     sensorAnalog)

/* This example checks the range of two attached laser rangefinders.
 * The collected data is then printed to the Debug Stream in real-time. */

task main()
{
  int leftValue;
  int rightValue;

  while(true)
  {
    leftValue = SensorValue[leftLaser];       //Get the left value.
    rightValue = SensorValue[rightLaser];     //Get the right value.
    writeDebugStream("[ %d , ", leftValue);   //Show the left value.
    writeDebugStreamLine("%d ]", rightValue); //Show the right value.
    wait1Msec(100);
  }
}
